codebase-pipelines/tests/unit.tftest.hcl

```diff
@@ -0,0 +1,253 @@
+mock_provider "aws" {}
+
+override_data {
+  target = data.aws_iam_policy_document.assume_codebuild_role
+  values = {
+    json = "{\"Sid\": \"AssumeCodebuildRole\"}"
+  }
+}
+
+override_data {
+  target = data.aws_iam_policy_document.codebuild_logs
+  values = {
+    json = "{\"Sid\": \"CodeBuildLogs\"}"
+  }
+}
+
+override_data {
+  target = data.aws_iam_policy_document.ecr_access
+  values = {
+    json = "{\"Sid\": \"ECRAccess\"}"
+  }
+}
+
+override_data {
+  target = data.aws_iam_policy_document.codestar_connection_access
+  values = {
+    json = "{\"Sid\": \"CodeStarConnectionAccess\"}"
+  }
+}
```

Comment:
I know we can't assert on the detail on the policy documents in the tests, but we can at least assert that the policy documents have been used in the correct places.

https://github.com/uktrade/terraform-platform-modules/pull/271/files#diff-8494f4b8e2b47c589035d642f9c28ffa867e9c9a25dc0055f6b9fc4928796860R856-R872

Reply:
It seems this only works with `aws_iam_policy` and not `aws_iam_role_policy`. The environment pipelines use both https://github.com/uktrade/terraform-platform-modules/blob/main/environment-pipelines/tests/unit.tftest.hcl#L606

Reply:
No I tried, it gives the "this cannot be tested on a plan error"